src/Service/HooksManager.php

```diff
@@ -44,4 +47,51 @@ public function updateAllowedProtocols($defaultProtocols)
 
         return array_merge($defaultProtocols, $customProtocols);
     }
+
+    /**
+     * Proxies requested asset files through PHP to serve them securely.
+     *
+     * @return void
+     */
+    public function proxyFile()
+    {
+        if (isset($_GET['assets'])) {
+            $asset             = sanitize_text_field($_GET['assets']);
+            $hashedAssetsArray = Option::getOptionGroup('hashed_assets', null, []);
+            $originalFilePath  = $this->getOriginalFilePath($asset, $hashedAssetsArray);
+
+            if ($originalFilePath && file_exists($originalFilePath)) {
+                header('Content-Type: application/javascript');
```

Comment:
This is NOT good, whole of this functionality should handle into the `AssetNameObfuscator` like

```
function proxyFile()
{
    if (isset($_GET['assets'])) {
        $asset = sanitize_text_field($_GET['assets']);
        $AssetNameObfuscator = new AssetNameObfuscator($asset);
        $AssetNameObfuscator->serveAsset();
    }
}
```

Or

```
function proxyFile()
{
    if (isset($_GET['assets'])) {
        $asset = sanitize_text_field($_GET['assets']);
        AssetNameObfuscator::serveAssetByHash($asset);
    }
}
```